### ISSUE TITLE
tgstat.cpp: error: 'unique_ptr' was not declared in this scope

### ISSUE THREAD

**Issue by barracuda156** [tanaylab/tgstat#9]
```
tgstat.cpp: In static member function 'static void TGStat::get_open_fds(std::set<int>&)':
tgstat.cpp:583:5: error: 'unique_ptr' was not declared in this scope
  583 |     unique_ptr<char[]> buf;
      |     ^~~~~~~~~~
tgstat.cpp:43:1: note: 'std::unique_ptr' is defined in header '<memory>'; did you forget to '#include <memory>'?
   42 | #include "tgstat.h"
  +++ |+#include <memory>
   43 | 
tgstat.cpp:583:16: error: expected primary-expression before 'char'
  583 |     unique_ptr<char[]> buf;
      |                ^~~~
tgstat.cpp:596:13: error: 'buf' was not declared in this scope
  596 |             buf = unique_ptr<char[]>(new char[fds_size]);
      |             ^~~
tgstat.cpp:596:30: error: expected primary-expression before 'char'
  596 |             buf = unique_ptr<char[]>(new char[fds_size]);
      |                              ^~~~
tgstat.cpp:599:71: error: 'buf' was not declared in this scope
  599 |                 buf_used = proc_pidinfo(getpid(), PROC_PIDLISTFDS, 0, buf.get(), fds_size);
      |                                                                       ^~~
tgstat.cpp:613:56: error: 'buf' was not declared in this scope
  613 |     struct proc_fdinfo *fdinfo = (struct proc_fdinfo *)buf.get();
      |                                                        ^~~
make: *** [tgstat.o] Error 1
```

**Comment by aviezerl** [comment]
1. Which OS are you using? 
2. Can you try adding the following line to `src/tgstat.h` and see if the problem persists? 

```cpp
#include <memory>
```
Alternatively, you can try installing from the following branch:

```r
remotes::install_github("tanaylab/tgstat", ref = "fix@memory-header")
```

**Comment by barracuda156** [comment]
@aviezerl Thank you for responding! Adding a header fixes the problem, we have already done that locally: https://github.com/macports/macports-ports/blob/master/R/R-tgstat/files/patch-fix-includes.diff

I am using 10.6 ppc with gcc, but we have confirmed successful builds on x86_64 and aarch64 systems with clang too.

P. S. There is one more fix needed for < 10.11: MAP_ANONYMOUS should be defined to MAP_ANON (conditionally), otherwise build fails, see: https://build.macports.org/builders/ports-10.10_x86_64-builder/builds/219243/steps/install-port/logs/stdio
Like here: https://github.com/macports/macports-legacy-support/blob/2806f66dbd0a7a47fb97fda5f9caedbac4b5f18b/include/sys/mman.h#L27-L31

PR with a simple fix: https://github.com/tanaylab/tgstat/pull/10 (condition on undef rather than specific macOS: simpler code, potential benefit for other systems)
